vizro-core/src/vizro/models/_navigation/nav_item.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import itertools
+from typing import Optional
+
+import dash
+import dash_bootstrap_components as dbc
+from dash import html
+from pydantic import validator
+
+from vizro._constants import STATIC_URL_PREFIX
+from vizro.models import VizroBaseModel
+from vizro.models._models_utils import _log_call
+from vizro.models._navigation._navigation_utils import _validate_pages
+from vizro.models._navigation.accordion import Accordion
+from vizro.models.types import NavPagesType
+
+
+class NavItem(VizroBaseModel):
+    """Icon to be used in Navigation Panel of Dashboard.
+
+    Args:
+        tooltip (Optional[str]): Text to be displayed in the tooltip on icon hover.
+        icon (Optional[str]): URI (relative or absolute) of the embeddable content.
+        pages (NavPagesType): See [NavPagesType][vizro.models.types.NavPagesType].
+    """
+
+    tooltip: Optional[str]
+    icon: Optional[str]
+    pages: NavPagesType
+    selector: Optional[Accordion] = None
+    text: Optional[str] = ""
+    max_text_length: int = 9
+
+    # Re-used validators
+    _validate_pages = validator("pages", allow_reuse=True, always=True)(_validate_pages)
+
+    @validator("selector", pre=True, always=True)
+    def set_selector(cls, selector, values):
+        if selector is None:
+            return Accordion(pages=values.get("pages"))
+        return selector
+
+    @validator("icon", always=True, pre=True)
+    def set_icon(cls, icon):
+        if icon is None:
+            return STATIC_URL_PREFIX + "/images/icon_1.svg"
+        return icon if icon else None
+
+    @_log_call
+    def pre_build(self):
+        if self.tooltip is None:
+            if self.text and len(self.text) > self.max_text_length:
+                self.tooltip = self.text
+                self.text = None
```

Comment:
Belongs in a validator for me - essentially everything where to change the default of a field I would move it to a validator if possible. Sometimes that's not the case e.g. if you need access to the model manager, you can only have it in the pre-build. But in this case, you only depend on other fields, so I would definitely prefer the validator. 

Here it's also becoming clear why the order of the fields matter e.g. if you want to validate tooltip, but you need access to text, you need to define text before tooltip in your model fields as validation is done by the order of these arguments listed in the model

Reply:
Implemented in root_validator here: https://github.com/mckinsey/vizro/pull/70/commits/222b583617a448a7162ac4339e3bae5f89a6e89d